examples/intervention_mode/README.md

```diff
@@ -0,0 +1,253 @@
+
+In your Python dependency manager, you will need to ensure these packages are installed. Afterwards, you should be able to run this as a regular Inspect eval.
+
+```
+$ inspect eval intervention.py --model openai/gpt-4o-2024-08-06
+```
+
+## Inspect Code
+
+### The Task
+
+At the bottom of `intervention.py`, we have our Inspect task:
+
+```python
+@task
+def intervention():
+    tools = [bash(), python()]
+    return Task(
+        dataset=MemoryDataset([Sample(input="Unused in interactive mode")]),
+        plan=[
+            system_message(SYSTEM_PROMPT),
+            agent_loop(tools),
+        ],
+        sandbox="docker",
+    )
+```
+
+As you can see, this is just a regular Inspect evaluation. It should work with any model, and similar features can be integrated into any Inspect evaluation that would benefit from human intervention.
+
+A few things to note here:
+
+* To add more tools, just put them in the `tools` list.
+* You may want to customize the system prompt, which can be found above the task definition.
+* It uses an agent loop using the [lower level `model.generate()` API.](https://inspect.ai-safety-institute.org.uk/agents-api.html)
+
+### The Agent Loop
+
+The agent loop is [just a plain Inspect solver.](https://inspect.ai-safety-institute.org.uk/solvers.html).
+
+First, we ask the user to enter a prompt for the model, print it out, and append it as a user message. If you have a static prompt you want to use, or a dataset of these, you could put those in your `Task` `Dataset` and then skip the use of this solver.
+
+```python
+@solver
+def get_user_prompt() -> Solver:
+    async def solve(state: TaskState, generate: Generate) -> TaskState:
+        with input_screen() as console:
+            user_prompt = Prompt.ask("Please enter your initial prompt for the model\n")
+            console.print(
+                Panel.fit(user_prompt, title="User Action", subtitle="Initial prompt")
+            )
+        state.messages.append(ChatMessageUser(content=user_prompt))
+
+        return state
+
+    return solve
+```
+
+Next, we have our agent loop. It runs a generation and appends the output to the messages list in `TaskState`. We also print the response of the model along with any tool calls present.
+
+If there are tool calls present, it will ask the user for explicit permission to run the tool calls. If this is granted, they will be run. The tool call output will then be appended to messages and printed to the screen.
+
+If there are no tool calls, the user will get a choice to terminate the conversation, force another generation, or send a custom user message to the model.
+
+```python
+@solver
+def agent_loop(tools: list[Tool]) -> Solver:
+    async def solve(state: TaskState, generate: Generate) -> TaskState:
+        model = get_model()
+        while True:
+            output = await model.generate(state.messages, tools)
```

Comment:
You need to assign `state.output` to ensure that the final completion is available for scoring.

Reply:
Assigned it right after this line